### ISSUE TITLE
Potential false-positive of Security / Hostname Substitution inspection

### ISSUE THREAD

**Issue by ihorvorotnov** [kalessil/phpinspectionsea#1163]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Bug/False-Positive |
| Plugin         | Php Inspections (EA Extended) v3.0.10 |
| Language level | PHP 7.2 |

Current behaviour (description/screenshot:)
--

```php
class GeoRedirect
{
    /**
     * @var array Production domain names.
     */
    private static $hosts = [
        'eu'    => 'www.sitename.eu',
        'co.uk' => 'www.sitename.co.uk',
        'com'   => 'sitename.com',
    ];

    /**
     * @var string The host we're currently on.
     */
    private $source_host;

    /**
     * Set up internal parameters according to request.
     */
    private function setup_params() : void
    {
        // Set source domain, if it's not faked.
        if ( in_array( $_SERVER['HTTP_HOST'], self::$hosts, true ) ) {
            $this->source_host = $_SERVER['HTTP_HOST'];
        }
    }
}
```

I'm checking the HTTP_HOST against a whitelist, and setting it only if it matches one of the values (there's a default value, but that's out of the scope of this issue). But the inspection "**Hostname Substitution**" still highlights the assignment as error.

<img width="948" alt="screen shot 2019-01-10 at 15 06 20" src="https://user-images.githubusercontent.com/1066083/50975560-983aa000-14f6-11e9-97ea-76ac4820467f.png">

Expected behaviour
--

The inspection should understand that an assignment happens only if the value is whitelisted, thus is safe.

Environment details
---

PhpStorm 2018.3.3
Build #PS-183.5153.36, built on January 9, 2019
JRE: 1.8.0_152-release-1343-b26 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.2

**Comment by kalessil** [comment]
Fixed!
